qiskit/transpiler/preset_passmanagers/level2.py

```diff
@@ -254,8 +254,14 @@ def _unroll_condition(property_set):
         sched = plugin_manager.get_passmanager_stage(
             "scheduling", scheduling_method, pass_manager_config, optimization_level=2
         )
-    init = common.generate_error_on_control_flow(
-        "The optimizations in optimization_level=2 do not yet support control flow."
+    init = common.generate_control_flow_options_check(
+        layout_method=layout_method,
+        routing_method=routing_method,
+        translation_method=translation_method,
+        optimization_method=optimization_method,
+        scheduling_method=scheduling_method,
+        basis_gates=basis_gates,
+        target=target,
```

Comment:
Under what conditions does it error now? I assume the check is still needed to avoid some stage options in `transpile()` which don't support control flow yet. I'm just worried that this will potentially block plugins which may support control flow operations.

Reply:
This checks if the user has requested one of our built-ins that does not currently support control flow. So for layout, that's `noise_adaptive`, for routing `loohahead` and `basic`, for translator `synthesis`, and for schedule both `alap` and `asap`. Any other pass names we allow since they'd be supplied by plugins.

Reply:
Perhaps not, though I'm not certain:

https://github.com/Qiskit/qiskit-terra/blob/f5f58522d10e1771f822c2cf57f0c73cf2de5a41/qiskit/transpiler/preset_passmanagers/common.py#L69-L71

@jakelishman is your comment from above resolved once we've got #10355 merged?

Reply:
Yeah, it's the collect 1q/2q runs and subsequent consolidation that was the problem at the time of this comment.

Reply:
Then, should we remove that from the error raise as part of this PR, since it should work with #10355

Reply:
Sounds good, I'll go ahead and remove the guard.

Reply:
Done in 2a2ae22.